gluten-core/src/main/scala/org/apache/gluten/extension/columnar/rewrite/RewriteSparkPlanRulesManager.scala

```diff
@@ -67,12 +67,16 @@ class RewriteSparkPlanRulesManager private (rewriteRules: Seq[RewriteSingleNode]
     }
   }
 
+  private val nodeNameMap = Map(
+    "SortMergeJoin" -> "ShuffledHashJoin",
+    "SortMergeJoin(skew=true)" -> "ShuffledHashJoin(skew=true)")
+
```

Comment:
Why is this mapping needed? This is like an exception comparing to other rewrites.

Reply:
year, other rules are to add nodes, and rewriteJoin is to replace nodes.

Reply:
cc @ulysses-you

Reply:
Use nodeName to get original plan is bit strict. It seems the only added node is ProjectExec. We can skip all ProjectExec and make sure there is only one target rewritten plan. e.g.:
```diff
-  private def getTransformHintBack(
-      origin: SparkPlan,
-      rewrittenPlan: SparkPlan): Option[TransformHint] = {
-    // The rewritten plan may contain more nodes than origin, here use the node name to get it back
+  private def getTransformHintBack(rewrittenPlan: SparkPlan): Option[TransformHint] = {
+    // The rewritten plan may contain more nodes than origin, for now it should only be
+    // `ProjectExec`.
     val target = rewrittenPlan.collect {
-      case p if p.nodeName == origin.nodeName => p
+      case p if !p.isInstanceOf[ProjectExec] => p
```

Reply:
Updated, also excludes `RewrittenNodeWall`.